Carrito.js

```diff
@@ -0,0 +1,132 @@
+export class Carrito {
+  constructor() {
+    // Recuperar el carrito desde localStorage, si existe, o inicializar como array vacío
+    this.items = JSON.parse(localStorage.getItem('carrito')) || [];
+  }
+
+  // Agregar un producto al carrito
+  agregarProducto(producto) {
+    // Buscar si el producto ya existe en el carrito
+    const productoExistente = this.items.find(item => item.name === producto.name);
+
+    if (productoExistente) {
+      // Si existe, incrementar la cantidad
+      productoExistente.quantity++;
+    } else {
+      // Si no existe, agregarlo al array de items con cantidad 1
+      this.items.push({ ...producto, quantity: 1 });
+    }
+
+    // Actualizar el localStorage después de agregar el producto
+    this.actualizarLocalStorage();
+  }
+
+  // Vaciar el carrito
+  vaciarCarrito() {
+    this.items = [];  // Limpiar el array de productos
+    this.actualizarLocalStorage();
+  }
+
+  // Eliminar un producto del carrito
+  eliminarProducto(productoName) {
+    this.items = this.items.filter(item => item.name !== productoName);  // Filtrar el producto a eliminar
+    this.actualizarLocalStorage();
+  }
+
+  // Incrementar la cantidad de un producto
+  aumentarCantidad(productoName) {
+    const producto = this.items.find(item => item.name === productoName);
+    if (producto) {
+      producto.quantity++;
+      this.actualizarLocalStorage();
+    }
+  }
+
+  // Disminuir la cantidad de un producto
+  disminuirCantidad(productoName) {
+    const producto = this.items.find(item => item.name === productoName);
+    if (producto && producto.quantity > 1) {
+      producto.quantity--;
```

Comment:
Esto tendría que hacerse por medio de un metodo propio de Producto, ocultando la implementación
```suggestion
      producto.decrementarCantidad();
```